crates/aptos-telemetry/src/sender.rs

```diff
@@ -48,16 +49,64 @@ pub(crate) struct TelemetrySender {
 }
 
 impl TelemetrySender {
-    pub fn new(base_url: &str, chain_id: ChainId, node_config: &NodeConfig) -> Self {
+    pub fn new(base_url: String, chain_id: ChainId, node_config: &NodeConfig) -> Self {
         Self {
-            base_url: base_url.into(),
+            base_url,
             chain_id,
             peer_id: node_config.peer_id().unwrap_or(PeerId::ZERO),
             client: reqwest::Client::new(),
             auth_context: Arc::new(AuthContext::new(node_config)),
         }
     }
 
+    pub(crate) async fn push_prometheus_metrics(&self) -> Result<(), anyhow::Error> {
+        debug!("Sending Prometheus Metrics");
+
+        let token = self.get_auth_token().await?;
+
+        let scraped_metrics =
+            prometheus::TextEncoder::new().encode_to_string(&aptos_metrics_core::gather())?;
+
+        let mut gzip_encoder = GzEncoder::new(Vec::new(), Compression::default());
+        gzip_encoder.write_all(scraped_metrics.as_bytes())?;
+        let compressed_bytes = gzip_encoder.finish()?;
+
+        let response = self
+            .client
+            .post(format!("{}/push-metrics", self.base_url))
+            .header("Content-Encoding", "gzip")
+            .bearer_auth(token)
+            .body(compressed_bytes)
+            .send()
+            .await;
+
+        match response {
+            Err(e) => Err(anyhow!("Prometheus Metrics push failed: {}", e)),
+            Ok(response) => {
+                if response.status().is_success() {
+                    debug!("Prometheus Metrics pushed successfully.");
+                    Ok(())
+                } else {
+                    Err(anyhow!(
+                        "Prometheus Metrics push failed with response: {}, body: {}",
+                        response.status(),
+                        response
+                            .text()
+                            .await
+                            .unwrap_or_else(|_| "empty body".to_string()),
+                    ))
+                }
+            }
+        }
+    }
+
+    pub(crate) async fn try_push_prometheus_metrics(&self) {
+        self.push_prometheus_metrics().await.map_or_else(
```

Comment:
We should add retry support here? If there is an epoch change, the service will unauthorize the request.